File: gcj2013R1CA.hs
#!/usr/bin/env runghc
main = getContents >>= mapM_ putStrLn . zipWith caseno [1..] . solve . lines
caseno i r = "Case #"++show i++": "++r
solve (c:cs) = map (toString . answer) . take (read c) . analyze $ cs ::[String]
analyze = map ((\[l,n]->(l,read n::Int)) . words)
toString = show
answer (l,n) = cnt (0,(-1,0)) . zip [0..] $ l where
  m = length l
  cnt (s,_) [] = s
  cnt (s,(b,cc)) ((i,c):xs)
    | c `elem` "aeiou" = cnt (s,(b,0)) xs
    | cc+1<n = cnt (s,(b,cc+1)) xs
    | otherwise = cnt (s+(i-n-b+1)*(m-i),(i-n+1,cc+1)) xs

File: gcj2013R1CA.py
#!/usr/bin/env python
import sys

def solve():
    ln = (sys.stdin.readline().split())
    L,M,N = ln[0], len(ln[0]), int(ln[1])
    def cnt((s,b,cc),(i,c)):
        if c in "aeiou": return (s,b,0)
        if cc<N-1: return (s,b,cc+1)
        return (s+(i-N-b+1)*(M-i),i-N+1,cc)
    return reduce(cnt,enumerate(L),(0,-1,0))[0]

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))
